src/Symfony/Component/Mailer/EventListener/SmimeEncryptedMessageListener.php

```diff
@@ -34,8 +35,19 @@ public function onMessage(MessageEvent $event): void
         if (!$message instanceof Message) {
             return;
         }
+        $certificates = [];
+        foreach ($event->getEnvelope()->getRecipients() as $recipient) {
+            if (!$recipient instanceof SMimeRecipient) {
+                continue;
+            }
+            $certificates[] = $recipient->certificatePath;
+        }
+        if (0 === \count($certificates)) {
```

Comment:
what if some recipients are not SMimeRecipient ?

Reply:
Indeed, an exception should be thrown as we cannot mix encrypted/normal emails.